src/include/duckdb/storage/buffer/buffer_pool.hpp

```diff
@@ -85,10 +88,35 @@ class BufferPool {
 	void IncrementDeadNodes(FileBufferType type);
 
 protected:
+	struct MemoryUsageCounters {
+		//! The maximum difference between memory statistics and actual usage is 2MB (64 * 32k)
+		static constexpr size_t kCacheCnts = 64;
+		static constexpr size_t kCacheThreshold = 32 << 10;
+		using MemoryUsagePerTag = std::array<atomic<int64_t>, MEMORY_TAG_COUNT>;
+
+		//! global memory usage counters
+		atomic<int64_t> memory_usage;
+		MemoryUsagePerTag memory_usage_per_tag;
+		//! cache memory usage to improve performance
+		std::array<MemoryUsagePerTag, kCacheCnts> memory_usage_caches;
```

Comment:
nit: `std::` is not necessary here when including `duckdb/common/array`, could you remove it?